pandas/core/frame.py

```diff
@@ -3263,6 +3263,8 @@ def _setitem_array(self, key, value):
             key = check_bool_indexer(self.index, key)
             indexer = key.nonzero()[0]
             self._check_setitem_copy()
+            if isinstance(value, DataFrame):
```

Comment:
does series hit this path?

Reply:
No, I did this because value could be an array for example

Reply:
ok great